### ISSUE TITLE
SSD300 と SSDLite の forward 実装を基底クラスに共通化

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#460]
## Summary

`SSD300Model` と `SSDLiteModel` の `forward` メソッドがほぼ完全に重複している. 基底クラス `SSDModelBase` を導入し共通化する.

## Background

両モデルは torchvision の異なる SSD 実装 (`ssd300_vgg16` / `ssdlite320_mobilenet_v3_large`) をラップしているが, 前向き推論と loss 計算の骨格はほぼ同一. 背景クラスのマスク処理も同じロジックで書かれている. 重複コードにより, 一方の修正が他方に波及しないリスクがある.

## Current Implementation

- `pochidetection/models/ssd300.py:59-105`
- `pochidetection/models/ssdlite.py:62-108`

差異はほぼ `_create_model` 相当の内部モデル生成のみ. forward 処理, pred_boxes / pred_logits / labels 構築, fg_mask 適用, loss 辞書の構造は全て同じ.

## Proposed Changes

- `pochidetection/models/ssd_base.py` (または既存適所) に `SSDModelBase(IDetectionModel)` を新設:

```python
class SSDModelBase(IDetectionModel):
    def __init__(self, num_classes: int, nms_iou_threshold: float) -> None:
        self._model: SSD = self._create_model(num_classes)
        self._num_classes = num_classes
        self._nms_iou_threshold = nms_iou_threshold

    @abstractmethod
    def _create_model(self, num_classes: int) -> SSD:
        """torchvision の SSD インスタンスを生成する."""

    def forward(self, ...):
        # 共通実装
```

- `SSD300Model` / `SSDLiteModel` は `_create_model` のみ実装し, `ssd300_vgg16` / `ssdlite320_mobilenet_v3_large` を返す.
- `nms_iou_threshold` のデフォルト値差異 (0.5 vs 0.55) は各サブクラスの `__init__` デフォルト値で吸収する.
- `label_to_torchvision` / `label_from_torchvision` (0-indexed ↔ 1-indexed 変換) も共通化できれば統合する.

## Acceptance Criteria

- [ ] `SSDModelBase` 基底クラスを導入する
- [ ] `SSD300Model` が `SSDModelBase` を継承する形に書き換える
- [ ] `SSDLiteModel` が `SSDModelBase` を継承する形に書き換える
- [ ] 既存のモデルテスト (`tests/test_models/`) が全て通る
- [ ] `uv run pochi train -c configs/ssd_coco.py --epochs 1` 相当のスモーク確認が通る (手動)

## Labels

`refactor`

## Branch

`refactor/ssd-model-base-class`
